include/convex_bodies/zpolytope.h

```diff
@@ -477,6 +477,35 @@ class Zonotope {
     {
         return line_intersect_coord(r, rand_coord, lamdas);
     }
+    
+
+    //------------------------------oracles for exponential sampling---------------//////
+
+    // compute intersection points of a ray starting from r and pointing to v
+    // with polytope discribed by A and b
+    std::pair<NT, int> quadratic_positive_intersect(Point const& r,
+                                    Point const& v,
+                                    VT const& Ac,
+                                    NT const& T,
+                                    VT& Ar,
+                                    VT& Av,
+                                    int& facet_prev) const
+    {
+        return std::make_pair(0, 0);
+    }
+
+    std::pair<NT, int> quadratic_positive_intersect(Point const& r,
+                                    Point const& v,
+                                    VT const& Ac,
+                                    NT const& T,
+                                    VT& Ar,
+                                    VT& Av,
+                                    NT const& lambda_prev,
+                                    int& facet_prev) const
+    {
+        return std::make_pair(0, 0);
```

Comment:
Should it be better if you threw an std::exception here?

Reply:
Yes sure. Done.